cmd/vulcan-vulners/main.go

```diff
@@ -265,20 +278,28 @@ func analyzeReport(target string, nmapReport *gonmap.NmapRun) ([]report.Vulnerab
 					continue
 				}
 				summary := fmt.Sprintf(vulnersVuln.Summary, port.Service.Product)
-				v, ok := uniqueVulns[summary]
+				uniqueVulnId := fmt.Sprintf("CPE: %s, Port: %d/%s", string(cpe), port.PortId, port.Protocol)
+				var cves []string
+
+				for _, row := range f.Resources.Rows {
+					cves = append(cves, row["CVE"])
+				}
+				v, ok := uniqueVulns[uniqueVulnId]
 				if !ok {
 					v.Vuln = report.Vulnerability{
-						Summary:         summary,
-						Description:     fmt.Sprintf(vulnersVuln.Description, port.Service.Product),
-						Recommendations: vulnersVuln.Recommendations,
+						ID:               computeVulnerabilityID(cves),
```

Comment:
We added a new field to store the fingerprint of the vulnerability that is being calculated here. The field is called `Fingerprint`, so we need to use it instead of the `ID` attribute.

Apart from that, I'd suggest also adding the `Score` of the CVEs, because if one of the CVEs is re-scored we might want to reopen the issue and let the user decide again.